### PR TITLE
refactor: replaces spec address with go-address 

### PR DESCRIPTION
**Pull request by frrist**
Replace address code with go-address and update spec to use the semantics from go-address.

Whats left TODO:
- Handle miner worker keys/addresses that use different protocols (i.e. BLS & SECP) called out in a comment. I propose we handle this in a follow on.

### CODE REVIEW

File: src/systems/filecoin_mining/storage_mining/storage_mining_subsystem.go [hunk 1]
```diff
@@ -288,12 +288,10 @@ func (sms *StorageMiningSubsystem_I) VerifyElectionPoSt(inds indices.Indices, he
...
-	// get worker key from minerAddr
-	workerKey, err := sma.Info().Worker().GetKey()
-	if err != nil {
-		return false
-	}
-
+	// TODO if the workerAddress is secp then payload will be the blake2b hash of its public key
+	// and we will need to recover the entire public key from worker before handing off to Verify
+	// example of recover code: https://github.com/ipsn/go-secp256k1/blob/master/secp256.go#L93
+	workerKey := sma.Info().Worker().Payload()
```
Comment (frrist) [inline src/systems/filecoin_mining/storage_mining/storage_mining_subsystem.go]: This code (and the below bits with a TODO) will not be executed anytime soon, but flagging that we will need to account for different key types before executing it.

File: src/actors/builtin/init/init_actor.go [hunk 2]
```diff
@@ -61,7 +62,8 @@ func (a *InitActorCode_I) Exec(rt Runtime, execCodeID abi.ActorCodeID, construct
 	})
 
 	return rt.ValueReturn(
-		Bytes(addr.Serialize_Address_Compact(idAddr)))
+		Bytes(idAddr.Bytes()),
```
Comment (zixuanzh) [inline src/actors/builtin/init/init_actor.go]: nit: idAddr.Bytes() should suffice

File: src/actors/builtin/storage_power/storage_power_actor.id [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import actor_util "github.com/filecoin-project/specs/actors/util"
-import addr "github.com/filecoin-project/specs/systems/filecoin_vm/actor/address"
+import addr "github.com/filecoin-project/go-address"
 import abi "github.com/filecoin-project/specs/actors/abi"
 
 type PowerTableHAMT {addr.Address: abi.StoragePower}  // TODO: convert address to ActorID
```
Comment (zixuanzh) [inline src/actors/builtin/storage_power/storage_power_actor.id]: noting that we should set key to type abi.ActorID as a follow up.

File: src/actors/builtin/init/init_actor.go [hunk 4]
```diff
@@ -1,13 +1,14 @@
...
 import (
+	addr "github.com/filecoin-project/go-address"
 	abi "github.com/filecoin-project/specs/actors/abi"
 	builtin "github.com/filecoin-project/specs/actors/builtin"
 	vmr "github.com/filecoin-project/specs/actors/runtime"
 	autil "github.com/filecoin-project/specs/actors/util"
 	ipld "github.com/filecoin-project/specs/libraries/ipld"
 	actor "github.com/filecoin-project/specs/systems/filecoin_vm/actor"
-	addr "github.com/filecoin-project/specs/systems/filecoin_vm/actor/address"
+	util "github.com/filecoin-project/specs/util"
```
Comment (anorth) [inline src/actors/builtin/init/init_actor.go]: Avoid depending on specs/util - you'll find equivalents in actors/util (or add what you need)

File: src/actors/builtin/init/init_actor.go (hunk 2 above)
Comment (anorth) [reply src/actors/builtin/init/init_actor.go]: Actually this should be the CBOR serialization.

File: src/actors/builtin/storage_miner/storage_miner_actor_code.go [hunk 5]
```diff
@@ -63,7 +64,7 @@ func (a *StorageMinerActorCode_I) OnSurprisePoStChallenge(rt Runtime) {
 	challengedSectorsRandomness := rt.RandomnessWithAuxSeed(
 		filcrypto.DomainSeparationTag_SurprisePoStSampleSectors,
 		rt.CurrEpoch()-node_base.SPC_LOOKBACK_POST,
-		addr.Serialize_Address_Compact(rt.CurrReceiver()),
+		rt.CurrReceiver().Bytes(),
```
Comment (anorth) [inline src/actors/builtin/storage_miner/storage_miner_actor_code.go]: Inputs to randomness are also defined to be CBOR, I believe. @sternhenri ?

File: src/actors/builtin/storage_miner/storage_miner_actor_code.go [hunk 6]
```diff
@@ -643,7 +644,7 @@ func (a *StorageMinerActorCode_I) _rtVerifySurprisePoStOrAbort(rt Runtime, onCha
 	postRandomnessInput := rt.RandomnessWithAuxSeed(
 		filcrypto.DomainSeparationTag_SurprisePoStVRFRandomnessInput,
 		challengeEpoch-node_base.SPC_LOOKBACK_POST,
-		addr.Serialize_Address_Compact(rt.CurrReceiver()),
+		rt.CurrReceiver().Bytes(),
```
Comment (anorth) [inline src/actors/builtin/storage_miner/storage_miner_actor_code.go]: CBOR

File: src/actors/builtin/storage_power/storage_power_actor_code.go [hunk 7]
```diff
@@ -12,8 +13,8 @@ import (
 	filcrypto "github.com/filecoin-project/specs/algorithms/crypto"
 	libp2p "github.com/filecoin-project/specs/libraries/libp2p"
 	sector "github.com/filecoin-project/specs/systems/filecoin_mining/sector"
-	addr "github.com/filecoin-project/specs/systems/filecoin_vm/actor/address"
 	ai "github.com/filecoin-project/specs/systems/filecoin_vm/actor_interfaces"
+	util "github.com/filecoin-project/specs/util"
```
Comment (anorth) [inline src/actors/builtin/storage_power/storage_power_actor_code.go]: Avoid specs/util

File: src/actors/runtime/runtime_util.go [hunk 8]
```diff
@@ -96,11 +96,13 @@ func RT_Address_Is_StorageMiner(rt Runtime, minerAddr addr.Address) bool {
 }
 
 func RT_GetMinerAccountsAssert(rt Runtime, minerAddr addr.Address) (ownerAddr addr.Address, workerAddr addr.Address) {
-	ownerAddr = addr.Deserialize_Address_Compact_Assert(
+	ownerAddr, err := addr.NewFromBytes(
```
Comment (anorth) [inline src/actors/runtime/runtime_util.go]: Deserialize CBOR

File: src/systems/filecoin_vm/runtime/impl/runtime.go [hunk 9]
```diff
@@ -664,15 +664,16 @@ func (rt *VMContext) RandomnessWithAuxSeed(
 
 func (rt *VMContext) NewActorAddress() addr.Address {
 	buf := new(bytes.Buffer)
-	_, err := buf.Write(addr.Serialize_Address(rt._immediateCaller))
+	_, err := buf.Write(rt._immediateCaller.Bytes())
```
Comment (anorth) [inline src/systems/filecoin_vm/runtime/impl/runtime.go]: Use CBOR here too for consistency

File: src/actors/builtin/storage_miner/storage_miner_actor_code.go (hunk 5 above)
Comment (frrist) [reply src/actors/builtin/storage_miner/storage_miner_actor_code.go]: I have changed this to use CBOR, is this comment merge blocking?

Comment (sternhenri) [reply src/actors/builtin/storage_miner/storage_miner_actor_code.go]: Picking this back up in #775. Simplified randomness for lotus purposes there

**Pull request merged**